src/coreclr/jit/hwintrinsiclistarm64sve.h

```diff
@@ -330,12 +330,33 @@ HARDWARE_INTRINSIC(Sve2,          ShiftArithmeticRounded,
+HARDWARE_INTRINSIC(Sve2,          ShiftRightArithmeticRoundedAdd,                                   -1,      3,     {INS_sve_srsra,      INS_invalid,        INS_sve_srsra,      INS_invalid,        INS_sve_srsra,      INS_invalid,        INS_sve_srsra,      INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_ShiftRightByImmediate, HW_Flag_Scalable|HW_Flag_HasImmediateOperand|HW_Flag_HasRMWSemantics)
+HARDWARE_INTRINSIC(Sve2,          ShiftRightArithmeticRoundedNarrowingSaturateEven,                 -1,      2,     {INS_sve_sqrshrnb,   INS_invalid,        INS_sve_sqrshrnb,   INS_invalid,        INS_sve_sqrshrnb,   INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_ShiftRightByImmediate, HW_Flag_Scalable|HW_Flag_HasImmediateOperand)
+HARDWARE_INTRINSIC(Sve2,          ShiftRightArithmeticRoundedNarrowingSaturateOdd,                  -1,      3,     {INS_sve_sqrshrnt,   INS_invalid,        INS_sve_sqrshrnt,   INS_invalid,        INS_sve_sqrshrnt,   INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_ShiftRightByImmediate, HW_Flag_Scalable|HW_Flag_HasImmediateOperand|HW_Flag_HasRMWSemantics)
+HARDWARE_INTRINSIC(Sve2,          ShiftRightArithmeticRoundedNarrowingSaturateUnsignedEven,         -1,      2,     {INS_invalid,        INS_sve_sqrshrunb,  INS_invalid,        INS_sve_sqrshrunb,  INS_invalid,        INS_sve_sqrshrunb,  INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_ShiftRightByImmediate, HW_Flag_Scalable|HW_Flag_HasImmediateOperand)
+HARDWARE_INTRINSIC(Sve2,          ShiftRightArithmeticRoundedNarrowingSaturateUnsignedOdd,          -1,      3,     {INS_invalid,        INS_sve_sqrshrunt,  INS_invalid,        INS_sve_sqrshrunt,  INS_invalid,        INS_sve_sqrshrunt,  INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_ShiftRightByImmediate, HW_Flag_Scalable|HW_Flag_HasImmediateOperand|HW_Flag_HasRMWSemantics)
+HARDWARE_INTRINSIC(Sve2,          ShiftRightLogicalAdd,                                             -1,      3,     {INS_invalid,        INS_sve_usra,       INS_invalid,        INS_sve_usra,       INS_invalid,        INS_sve_usra,       INS_invalid,        INS_sve_usra,       INS_invalid,        INS_invalid},     HW_Category_ShiftRightByImmediate, HW_Flag_Scalable|HW_Flag_HasImmediateOperand|HW_Flag_HasRMWSemantics)
+HARDWARE_INTRINSIC(Sve2,          ShiftRightLogicalNarrowingEven,                                   -1,      2,     {INS_sve_shrnb,      INS_sve_shrnb,      INS_sve_shrnb,      INS_sve_shrnb,      INS_sve_shrnb,      INS_sve_shrnb,      INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_ShiftRightByImmediate, HW_Flag_Scalable|HW_Flag_HasImmediateOperand)
```

Comment:
`shrnb` just operates on unsigned integer value. @a74nh  - do you know why we are exposing `ShiftRightLogicalNarrowingEven` for signed and unsigned? Similar argument goes for other APIs.

Reply:
Signed and unsigned versions exist in the ACLE: https://developer.arm.com/architectures/instruction-sets/intrinsics/#f:@navigationhierarchiessimdisa=[sve2]&q=svshrnb

I don't think the signedness of the input really matters, just that everything gets shifted right and narrowed.

If we just had one version, I can imagine it creating lots of casting when used with the wrong type. Similar to how were looking at adding additional type for SVE APIs we did last year.